packages/bpk-component-phone-input/src/BpkPhoneInput.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { cssModules } from 'bpk-react-utils';
+import BpkInput, { INPUT_TYPES } from 'bpk-component-input';
+import BpkSelect from 'bpk-component-select';
+
+import STYLES from './BpkPhoneInput.scss';
+
+const getClassName = cssModules(STYLES);
+
+type ChildProps = { [string]: any };
+
+export type Props = {
+  dialingCode: string,
+  dialingCodeProps: { id: string, name: string, className?: string },
+  dialingCodes: Array<{ code: string, description: string }>,
+  id: string,
+  name: string,
+  onChange: (string, string) => void,
+  value: string,
+  className: ?string,
+  containerProps: ChildProps,
+  disabled: ?boolean,
+  large: ?boolean,
+  valid: ?boolean,
+};
+
+type CommonProps = {
+  large: boolean,
+  disabled: ?boolean,
+  valid: ?boolean,
+};
+
+const BpkPhoneInput = (props: Props) => {
+  const {
+    id,
+    className,
+    name,
+    disabled,
+    onChange,
+    valid,
+    value,
+    large,
+    dialingCode,
+    dialingCodes,
+    dialingCodeProps,
+    containerProps,
+    ...rest
+  } = props;
+
+  const commonProps: CommonProps = {
+    valid,
+    large: !!large,
+    disabled,
+  };
+
+  return (
+    <span
+      {...containerProps}
+      className={getClassName('bpk-phone-input', containerProps.className)}
+    >
+      <BpkSelect
+        {...commonProps}
+        {...dialingCodeProps}
+        className={getClassName(
+          'bpk-phone-input__dialing-code',
+          dialingCodeProps.className,
+        )}
+        value={dialingCode}
+        onChange={(e: SyntheticInputEvent<HTMLElement>) =>
+          onChange(e.target.value, value)
+        }
+      >
+        {dialingCodes.map(({ code, description, ...extraDialingProps }) => (
+          <option key={code} value={code} {...extraDialingProps}>
+            {description}
+          </option>
+        ))}
+      </BpkSelect>
+      <BpkInput
+        {...commonProps}
+        {...rest}
```

Comment:
`...rest` should generally be placed on the outermost container except for some very specific circumstances

Reply:
Yeah, this is one of the those specific circumstances :D. When used with the FieldSet, it passes trough some a11y props that we need to forward to the input (to make it accessible). The options were to either filter all those properties and pass it to the input, which would be a bit confusing, or this. I talked with @matthewdavidson and we decided to go this way and have a property called `containerProps` so users can still pass in extra props to the container.

Reply:
Ah fair enough. We'll leave it as it is then. The only thing I might suggest is pulling `style` out and applying it to the outer container, as some consumers use style to adjust positioning etc